matrix/helper_functions.hpp

```diff
@@ -20,64 +20,57 @@ bool is_finite(Type x) {
 #endif
 }
 
+/**
+ * Wrap value to stay in range [low, high)
+ *
+ * @param x input possibly outside of the range
+ * @param low lower limit of the allowed range
+ * @param high upper limit of the allowed range
+ * @return wrapped value inside the range
+ */
 template<typename Type>
-Type wrap_pi(Type x)
-{
-    if (!is_finite(x)) {
+Type wrap(Type x, Type low, Type high) {
```

Comment:
Maybe we should have an integer overload version that uses the `%` operator? This is very fast on ARM...

Also, I would have phrased the API as `Type wrap(Type x, Type range, Type offset)`, so the wrap_2pi would be `wrap(x, M_TWOPI, 0)`, and wrap_pi would be `wrap(x, M_TWOPI, M_PI)`. But this one is fine too, just different ways of thinking about the problem.